api/tests/test_gdpr/test_gdpr_api.py

```diff
@@ -0,0 +1,159 @@
+            "aud": audience,
+            "sub": str(user.uuid),
+            "iat": int(now.timestamp()),
+            "exp": int(expire.timestamp()),
+            auth_field: scopes,
+        }
+        encoded_jwt = jwt.encode(
+            jwt_data, key=rsa_key.private_key_pem, algorithm=rsa_key.jose_algorithm
+        )
+
+        req_mock.get(config_url, json=configuration)
+        req_mock.get(jwks_url, json=keys)
+
+        auth_header = f"{api_token_auth_settings.AUTH_SCHEME} {encoded_jwt}"
+
+        return auth_header
+
+    @requests_mock.Mocker()
+    def test_get_user_data_should_return_data(self, req_mock):
+        auth_header = self.get_auth_header(
+            self.user, [settings.GDPR_API_QUERY_SCOPE], req_mock
+        )
+
+        self.client.credentials(HTTP_AUTHORIZATION=auth_header)
+        response = self.client.get(self.url)
+        assert_that(response.status_code).is_equal_to(200)
+        assert_that(response.data["children"][0]["value"]).is_not_empty()
+
+    @requests_mock.Mocker()
+    def test_get_user_data_should_not_returns_not_found(self, req_mock):
+        auth_header = self.get_auth_header(
+            self.user, [settings.GDPR_API_QUERY_SCOPE], req_mock
+        )
+
+        url = reverse("gdpr_v1", kwargs={"uuid": uuid.uuid4()})
+        self.client.credentials(HTTP_AUTHORIZATION=auth_header)
+        response = self.client.get(url)
+        assert_that(response.status_code).is_equal_to(404)
+
+    @requests_mock.Mocker()
+    def test_get_user_returns_forbidden_when_wrong_scope(self, req_mock):
+        auth_header = self.get_auth_header(self.user, ["testprefix.invalid"], req_mock)
+        self.client.credentials(HTTP_AUTHORIZATION=auth_header)
+        response = self.client.get(self.url)
+        assert_that(response.status_code).is_equal_to(403)
+
+    @requests_mock.Mocker()
+    def test_delete_user_should_anonymize_data(self, req_mock):
+        auth_header = self.get_auth_header(
+            self.user, [settings.GDPR_API_DELETE_SCOPE], req_mock
+        )
+        old_uuid = self.user.uuid
+        self.client.credentials(HTTP_AUTHORIZATION=auth_header)
+        response = self.client.delete(self.url)
+
+        assert_that(response.status_code).is_equal_to(204)
+
+        self.user.refresh_from_db()
+        assert_that(self.user.uuid).is_not_equal_to(old_uuid)
+        assert_that(self.user.username).contains("anonymized")
+        assert_that(self.user.pk).is_not_none()
+
+    @requests_mock.Mocker()
+    def test_delete_user_returns_forbidden_when_using_wrong_scope(self, req_mock):
+        auth_header = self.get_auth_header(
+            self.user, ["testprefix.wrong_scope"], req_mock
+        )
+        self.client.credentials(HTTP_AUTHORIZATION=auth_header)
+        response = self.client.delete(self.url)
+        assert_that(response.status_code, 403)
+        self.user.refresh_from_db()
+        assert_that(self.user.username).does_not_contain("anonymized")
+
```

Comment:
Should a test be added that tries to anonymize a user, that is not currently logged in? So that users cannot start anonymizing everyone they want destroying functionality.